src/spdx/parser/rdf/graph_parsing_functions.py

```diff
@@ -102,3 +102,44 @@ def remove_prefix(string: str, prefix: str) -> str:
     if string.startswith(prefix):
         return string[len(prefix) :]
     return string
+
+
+def get_correctly_typed_triples(
+    logger: Logger,
+    graph: Graph,
+    subject: Optional[Node] = None,
+    predicate: Optional[Node] = None,
+    _object: Optional[Node] = None,
+) -> Tuple[URIRef, Node, Union[BNode, Literal, URIRef]]:
+    # this is a helper method to enforce some rdf types for the triples to bo compatible with the
+    # code that follows
+    for s, p, o in graph.triples((subject, predicate, _object)):
+        if not isinstance(s, (BNode, URIRef)):
+            logger.append(
+                f"Warning: Subject should be of type BNode or URIRef, but is {type(s).__name__}. "
+                f"This might lead to a failure."
```

Comment:
We might want to include the actual node, else it will be hard to discern what actually caused the warning.

Reply:
Done